content/blog/jjpx/2018-09-05_haiku_monthly_activity_report_august_2018.md

```diff
@@ -0,0 +1,43 @@
++++
+type = "blog"
+title = "Haiku monthly activity report - 08/2018"
+author = "jjpx"
+date = "2018-09-05 16:37:40-03:00"
+tags = []
++++
+
+Time has come and a new monthly activity report arrived with it. It is the first time I write something this long in english. Enjoy!
+
+This report covers hrev52141-52310.
+
+### Drivers
+
+SDHCI MMC driver from [krish_iyer](https://www.haiku-os.org/blog/krish_iyer) has been merged in master. There is still a lot of work to do and it is not enabled in automated builds yet, but should serve as a basis for any future related work. His commits include extensive documentation on how to setup a testing vm, what has been done and what is left to do.
+
+There have been several contributions to the ARM port including work by kallisti5 on our implementation fo Flattened Device Tree (FDT) that should make it easier for Haiku to be run on more ARM hardware variants and a rewrite of the memcpy function for ARM devices by Augustin Cavalier based on Clang 8.
+
+Finally on the driver related news kallisti5 fixed an SMAP violation on s3 accelerant.
+
+### Fixes
+
+PulkoMandy fixed a bug in MediaPlayer related to playing streams of unknown length such as an online radio.
+
+phuodoin fixed a bug where the system became unresponsive when a textsearch is used in a big search.
+
+### Others
+
+waddlesplash fixed and enabled Reiserfs on x86_64 builds. Previously this filesystem was only available on x86 builds.
+
+Zstandard has been added as an alternative and high ratio compression algorithm for Haiku Packages. Although it is not the default, a quick test by diver reduced LibreOffice package from 112MB to 107MB with best compression.
```

Comment:
Again a bit of background info:

Haiku packages are currently compressed using zlib. This is a well-known and mature library, but it is not state of the art anymore. Korli added support for Facebook's Zstandard compression library, which allows packages of a slightly smaller size and more important, is faster at compression and decompression (in the case of Haiku this will have an impact on boot time).

Reply:
Let's say Zstd support is experimental. And because compression is per block, a better compression rate is difficult to achieve anyway.